File: src/main/java/com/skyspacefiftyfive/service/BoardingPassService.java
package com.skyspacefiftyfive.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspacefiftyfive.domain.BoardingPass;
import com.skyspacefiftyfive.dto.BoardingPassDTO;
import com.skyspacefiftyfive.dto.BoardingPassSearchDTO;
import com.skyspacefiftyfive.dto.BoardingPassPageDTO;
import com.skyspacefiftyfive.dto.BoardingPassConvertCriteriaDTO;
import com.skyspacefiftyfive.service.GenericService;
import com.skyspacefiftyfive.dto.common.RequestDTO;
import com.skyspacefiftyfive.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface BoardingPassService extends GenericService<BoardingPass, Integer> {

	List<BoardingPass> findAll();

	ResultDTO addBoardingPass(BoardingPassDTO boardingPassDTO, RequestDTO requestDTO);

	ResultDTO updateBoardingPass(BoardingPassDTO boardingPassDTO, RequestDTO requestDTO);

    Page<BoardingPass> getAllBoardingPasss(Pageable pageable);

    Page<BoardingPass> getAllBoardingPasss(Specification<BoardingPass> spec, Pageable pageable);

	ResponseEntity<BoardingPassPageDTO> getBoardingPasss(BoardingPassSearchDTO boardingPassSearchDTO);
	
	List<BoardingPassDTO> convertBoardingPasssToBoardingPassDTOs(List<BoardingPass> boardingPasss, BoardingPassConvertCriteriaDTO convertCriteria);

	BoardingPassDTO getBoardingPassDTOById(Integer boardingPassId);







}

File: skyspacefiftyfive_ui/src/services/BoardingPassService.js
import http from "../http-common"; 

class BoardingPassService {
  getAllBoardingPasss(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/boardingPass/boardingPasss`, searchDTO);
  }

  get(boardingPassId) {
    return this.getRequest(`/boardingPass/${boardingPassId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/boardingPass?field=${matchData}`, null);
  }

  addBoardingPass(data) {
    return http.post("/boardingPass/addBoardingPass", data);
  }

  update(data) {
  	return http.post("/boardingPass/updateBoardingPass", data);
  }
  
  uploadImage(data,boardingPassId) {
  	return http.postForm("/boardingPass/uploadImage/"+boardingPassId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new BoardingPassService();
